vassal-app/src/main/java/VASSAL/tools/image/GeneralFilter.java

```diff
@@ -690,9 +715,9 @@ private static void apply_v(
 
         dst[dx + i*dw] =
           a << 24 |
-          (r > a ? a : r) << 16 |
-          (g > a ? a : g) <<  8 |
-          (b > a ? a : b);
+          (Math.min(r, a)) << 16 |
+          (Math.min(g, a)) <<  8 |
+          (Math.min(b, a));
```

Comment:
Don't make code changes to `GeneralFilter` without benchmarking it. This is the most heavily-optimized code we have, and small changes like  this can have a surprisingly significant impact on tiling speed.

Reply:
Reverted